subliminal/core.py

```diff
@@ -357,6 +357,48 @@ def search_external_subtitles(path, directory=None):
     return subtitles
 
 
+def raise_for_invalid_file(path):
+    """Checks if is a valid file."""
+    # check for non-existing path
+    if not os.path.exists(path):
+        raise ValueError('Path does not exist')
+
+    # check that this is a file and not a directory
+    if not os.path.isfile(path):
+        raise ValueError('%r is not a file' % path)
+
+
+def get_video_file_from_rar(path):
+    """Return a tuple with video filename and size from rar."""
+    # make sure this is really a rar file
+    if not is_rarfile(path):
+        raise ValueError("'{0}' is not a valid archive".format(os.path.splitext(path)[1]))
+
+    rar = RarFile(path)
+
+    # check that the rar doesnt need a password
+    if rar.needs_password():
+        raise ValueError('Rar requires a password')
+
+    # raise an exception if the rar file is broken
+    # must be called to avoid a potential deadlock with some broken rars
+    rar.testrar()
+
+    # use infolist instead to filter out directories and return only video files, since namelist returns directories
+    file_info = [f for f in rar.infolist() if not f.isdir() and f.filename.endswith(VIDEO_EXTENSIONS)]
+
+    # sort by file size descending, the largest video in the archive is the one we want, there may be samples or intros
+    file_info.sort(key=operator.attrgetter('file_size'), reverse=True)
+
+    # no video found
+    if not file_info:
+        raise ValueError('No video in archive')
+
+    # Free the information about irrelevant files before guessing
+    file_info = file_info[0]
+    return file_info.filename, file_info.file_size
+
+
```

Comment:
It seems those two functions are called only once so there is no added value in creating separate functions, just put that in the `scan_video`, it will save some trouble for code readers to go back and forth from the function definition and the main `scan_video`.